document/CoreFormat.mdk

```diff
@@ -1006,71 +1006,99 @@ only the last element represents the actual file within which the result occurre
+
+~ IsoNote
+SARIF defines column number as a count of characters.
+If a line in a text file contains tab characters, viewers may choose to present column
+numbers that match the visual offset of each character from the beginning of the line.
+These "visual" column numbers will not match the column numbers contained in the SARIF file.
+~
+
+Each character may be represented by one byte or by multiple bytes, depending on the file's character encoding.
+In source files encoded in UTF-16, characters outside the Basic Multilingual Plane (BMP)
+are represented as a sequence of two 16-bit code points;
+this sequence is called a "surrogate pair."
+Tools that report results in UTF-16-encoded files shall consider characters outside the BMP
+as occupying two columns.
 
-### Regions in text files
+~ IsoNote
+The reason for this requirement is that is common for existing tools to ignore surrogate pairs
+when calculating column numbers.
+~
+
+~ IsoNote
+Programs such as viewers that process SARIF log files together with the analysis target files
+to which those log files refer should attempt to determine the character encoding of target files.
+In the absence of internal information such as a Byte Order Mark, viewers may use external information
+(for example, command line arguments, project settings, or other configuration information)
+to determine the character encoding.
+If external information is also lacking, viewers should assume that each character occupies one byte.
+~
 
-When the `physicalLocationComponent` object (&sect;[#physicalLocationComponent]) in which a `region` object occurs
-represents a text file,
-the `region` object represents a span that consists of a contiguous range of zero or more characters within that file.
-In that case, the `byteOffset` property (&sect;[#region-byteOffset]) shall not be present.
+The start of a region in a text file shall be represented by a combination of the `startLine` (&sect;[#region-startLine])
+and `startColumn` (&sect;[#region-startColumn]) properties.
+`startLine` shall be present.
+If `startColumn` is absent, the region shall be considered to start at column 1.
+For the remainder of this section, whenever `startColumn` is mentioned, it includes the case where `startColumn` was absent
+and so is considered to be 1.
 
-The start of a region in a text file shall be represented in one of the following ways:
+The end of a region in a text file shall be represented either by a combination of the `endLine`
+(&sect;[#region-endLine]) and `endColumn` (&sect;[#region-endColumn]) properties,
+or by the `length` property (&sect;[#region-length]).
 
-*	By a combination of the `startLine` (&sect;[#region-startLine]) and `startColumn` (&sect;[#region-startColumn]) properties.
-    If `startColumn` is not specified, the region is assumed to start at column 1.
+If either of `endLine` or `endColumn` is present, then `length` shall not be present.
```

Comment:
Do we need this restriction? and the one that follows? if it is clear elsewhere that we are working with a text region, why isn't ok for producers to aggressively fill out all this information?